### ISSUE TITLE
[Question] createEntityAdapter with normalized data

### ISSUE THREAD

**Issue by rhalaly** [reduxjs/redux-toolkit#646]
In your docs you've introduced this demo: https://codesandbox.io/s/rtk-entities-basic-example-with-normalizr-bm3ie
I understand that this demo is toy demo, but it was hard for me to generalize it.

You've created `fetchArticle` that fetch the articles, normalize them and every other entity implements the `fetchArticle.fulfilled` in thiers `extraReducers` section.

But lets say that I have a new function `featchUsers` that returns the list of users and the last comments of each user. In this case, the comments `extraReducer` has to implement both `fetchArticle.fulfilled` and `featchUsers.fullfilled` like:

```
export const slice = createSlice({
  name: "comments",
  initialState: commentsAdapter.getInitialState(),
  reducers: {},
  extraReducers: {
    [fetchArticle.fulfilled]: (state, action) => {
      commentsAdapter.upsertMany(state, action.payload.comments);
    },
    [fetchUsers.fulfilled]: (state, action) => {
      commentsAdapter.upsertMany(state, action.payload.comments);
    },
  }
});
```

So, we have some code duplication... Not so beautiful but we have only two...

But what if I have 10 different fetches that touch the comment entity after normalization? The `extraReducers` of the comments entity will has 10 duplicated `commentsAdapter.upsertMany(state, action.payload.comments);` methods.

So what is the best approach to handle normalization with many entities / fetches in the system?

**Comment by markerikson** [comment]
That bit of code looks fine to me.  If you _really_ want to avoid duplication, do:

```js
function upsertComments(state, action) {
  commentsAdapter.upsertMany(state, action.payload.comments);
}

export const slice = createSlice({
  name: "comments",
  initialState: commentsAdapter.getInitialState(),
  reducers: {},
  extraReducers: {
    [fetchArticle.fulfilled]: upsertComments,
    [fetchUsers.fulfilled]: upsertComments
  }
});
```

Alternately, using the new "matcher" syntax we introduced:

```js
export const slice = createSlice({
  name: "comments",
  initialState: commentsAdapter.getInitialState(),
  reducers: {},
  extraReducers: builder => {
    builder.addMatcher(
      action => !!action.payload.comments,
      upsertComments
    ); 
  }
});
```

which will match against any action where `action.payload.comments` is truthy and handle the insertion.

**Comment by rhalaly** [comment]
Thank you! The matcher syntax is exactly what I was looking for! Currently I see only `addCase` (I'm using TypeScript), so I will try to update the lib tomorrow and see if it there

**Comment by markerikson** [comment]
We just released that in v1.4.0:

https://github.com/reduxjs/redux-toolkit/releases/tag/v1.4.0
